### DISCUSSION TITLE
ipsec/openvpn routing

### DISCUSSION THREAD

**Discussion by ivansmm** [strongswan/strongswan#1296] [General]
Hallo!
I have openvpn and strongswan vpns running on the single server. Openvpn provides both client and site-to-site connections, strongswan handles some other site-to-site tunnels. The server is behind NAT, so strongswan uses udp encapsulation for nat traversal.
When I connect as openvpn client (using bridged mode) I'm trying to reach the location at the remote end of one of the ipsec tunnels. When I upload data to this remote location from client machine, the throughput is about 1Mb/s. But when I copy the data back, the throughput drops to mere 24Kb/s. When I download data from remote to vpn gateway directly (i.e. excluding openvpn client connection), throughput rises to 100Kb/s, but still does not reach upload speed.
When I look at the traffic capture I observe a lot of tcp data retransmissions and duplicate acks.
No such problem occurs when the traffic is routed through openvpn tunnel.
Moreover, this problem does not occur on routing through all ipsec tunnels I have, some of them do not excibit such significant difference in upload/download speeds.
Both openvpn and stronwswan are running on current debian stable (openvpn 2.5.1 and strongswan 5.9.1).
Can somebody give me an advice where to look to investigate this problem?
I observed that udp packets, used for ESP encapsulation, on slow ipsec tunnels are longer than on the fast one (1460 bytes on fast tunnel, 1464 and 1480 on slow). Is there a way to limit UPD packet size for ipsec?
Best regards!
Top

**Comment by tobiasbrunner** [comment]
The docs have some info [on MTU/MSS issues](https://docs.strongswan.org/docs/5.9/howtos/forwarding.html#_mtumss_issues).

**Reply by ivansmm** [reply]
Thanks, I have seen this doc. I tried what was suggested there, but without success. XfrmOutStateModeError is 0 at my side.

**Reply by Thermi** [reply]
Doesn't mean anything because the packet can get too large and be dropped outside.

**Reply by ivansmm** [reply]
The issue was finally resolved by turning off ipsec acceleration on remote router! :-) (sophos)
